src/webgpu/shader/execution/expression/call/builtin/length.spec.ts

```diff
@@ -9,28 +9,81 @@ Returns the length of e (e.g. abs(e) if T is a scalar, or sqrt(e[0]^2 + e[1]^2 +
 
 import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
 import { GPUTest } from '../../../../../gpu_test.js';
-import { allInputSources } from '../../expression.js';
+import { TypeF32, TypeVec } from '../../../../../util/conversion.js';
+import { lengthInterval } from '../../../../../util/f32_interval.js';
+import { fullF32Range, kVectorTestValues } from '../../../../../util/math.js';
+import {
+  allInputSources,
+  Case,
+  makeUnaryToF32IntervalCase,
+  makeVectorToF32IntervalCase,
+  run,
+} from '../../expression.js';
+
+import { builtin } from './builtin.js';
 
 export const g = makeTestGroup(GPUTest);
 
+/** @returns a `length` Case for a vector of f32s input */
+const makeCaseVecF32 = (x: number[]): Case => {
+  return makeVectorToF32IntervalCase(x, lengthInterval);
+};
+
 g.test('abstract_float')
-  .specURL('https://www.w3.org/TR/WGSL/#float-builtin-functions')
+  .specURL('https://www.w3.org/TR/WGSL/#numeric-builtin-functions')
   .desc(`abstract float tests`)
   .params(u =>
     u.combine('inputSource', allInputSources).combine('vectorize', [undefined, 2, 3, 4] as const)
   )
   .unimplemented();
 
 g.test('f32')
-  .specURL('https://www.w3.org/TR/WGSL/#float-builtin-functions')
+  .specURL('https://www.w3.org/TR/WGSL/#numeric-builtin-functions')
   .desc(`f32 tests`)
   .params(u =>
-    u.combine('inputSource', allInputSources).combine('vectorize', [undefined, 2, 3, 4] as const)
+    u.combine('inputSource', allInputSources)
   )
-  .unimplemented();
+  .fn(async t => {
+    const makeCase = (x: number): Case => {
+      return makeUnaryToF32IntervalCase(x, lengthInterval);
+    };
+    const cases: Case[] = fullF32Range().map(makeCase);
+
+    await run(t, builtin('length'), [TypeF32], TypeF32, t.params, cases);
```

Comment:
Did we decide that this needs to allow `x` and `x*x`?

Reply:
Yup, sqrt(x*x) for scalars, https://github.com/gpuweb/gpuweb/pull/3409